homeassistant/components/sanix/coordinator.py

```diff
@@ -0,0 +1,54 @@
+"""Sanix Coordinator."""
+from datetime import timedelta
+import logging
+
+from sanix import Sanix
+from sanix.exceptions import SanixException
+
+from homeassistant.core import HomeAssistant
+from homeassistant.helpers.update_coordinator import DataUpdateCoordinator, UpdateFailed
+
+from .const import (
+    ATTR_API_BATTERY,
+    ATTR_API_DEVICE_NO,
+    ATTR_API_DISTANCE,
+    ATTR_API_FILL_PERCENTAGE,
+    ATTR_API_SERVICE_DATE,
+    ATTR_API_SSID,
+    ATTR_API_STATUS,
+    ATTR_API_TIME,
+    MANUFACTURER,
+)
+
+_LOGGER = logging.getLogger(__name__)
+
+
+class SanixCoordinator(DataUpdateCoordinator):
+    """Sanix coordinator."""
+
+    def __init__(self, hass: HomeAssistant, sanix_api: Sanix) -> None:
+        """Initialize coordinator."""
+        super().__init__(
+            hass, _LOGGER, name=MANUFACTURER, update_interval=timedelta(hours=1)
+        )
+        self._sanix_api = sanix_api
+
+    async def _async_update_data(self) -> dict[str, str | float | int]:
```

Comment:
Also, since you are using `.get()`, the value can be `None`. Is this something you expect?

Second thing, maybe you can look into a dataclass, which gives you more typesafety.

Reply:
Okay, I have changed it a bit :) is this something you expected?